feat: Allow excluding files from UI5 Linter checks (#264)
SAP/ui5-linter: README.md

ui5lint --format json
```

#### `--ignore-pattern`

Pattern/files that will be ignored for linting. Can be defined also in `ui5lint.config.js`.

**Example:**
```sh

> ui5lint --ignore-pattern "./test"

Reviewer: Maybe use a more realistic pattern, such as `webapp/thirdparty/**`?

Author: Thanks!

```

## Configuration

> UI5 Linter can be easily configured using an external configuration file. This file allows you to customize how the linter behaves, such as ignoring specific files or directories. 

Reviewer 2: ```suggestion
The UI5 linter can easily be configured with an external configuration file, allowing you to customize how the linter behaves. For example, you can tell it to ignore specific files or directories. 
```

### Configuration File Location
The configuration file must be placed in the root directory of your project, alongside the `ui5.yaml` and `package.json` files. The linter will automatically detect and load the file when it runs.

### Supported Configuration File Names

> You can name your configuration file in one of the following formats:

Reviewer 2: ```suggestion
You can name your configuration file using one of the following formats:
```

- `ui5lint.config.js`
- `ui5lint.config.mjs`
- `ui5lint.config.cjs`

If you need to specify a custom configuration file, you can provide it using the `--config` parameter via the command line.

### Configuration File Format

#### ESM (ECMAScript Modules):
```js
export default {
  ignores: [
    "test/**/*", 

>     "!test/sap/m/visual/Wizard.spec.js",

Reviewer: Same here: Rather use an example that is more common for users (which usually have application projects)

Author: Now should be better :)

  ],
};
```

#### CommonJS:
```js
module.exports = {
  ignores: [
    "test/**/*", 
    "!test/sap/m/visual/Wizard.spec.js",
  ],
};
```

### Configuration Options

> - **ignores**: This option allows you to define glob patterns to ignore specific files or directories during linting. Patterns are relative to the root of the project. You can also un-ignore specific files by using the `!` prefix. The order of the patterns matters—later patterns can override earlier ones.

Reviewer 2: ```suggestion
- **ignores**: This option allows you to define glob patterns to ignore specific files or directories during linting. Patterns are relative to the root of the project. You can also un-ignore specific files by using the `!` prefix. The order of the patterns matters; later patterns override earlier ones.
```

Example:
```js
ignores: [
  "test/**/*",              // Ignore all files in the test folder
  "!test/sap/m/visual/**",  // Un-ignore files in a specific subdirectory
];
```

> This provides flexible control over which files the UI5 Linter should process and which it should ignore.

Reviewer 2: ```suggestion
In this way, you can control which files the UI5 linter should process and which it should ignore.
```
